From c1c7e9fccd76fd4d4e7594ccace12f0ad7ade026 Mon Sep 17 00:00:00 2001
From: Mengjun Leng <mengju@codeaurora.org>
Date: Tue, 22 Nov 2016 15:48:25 +0800
Subject: [PATCH] InCallUI: Add a controller for proximity sensor

Add the setting for enable proximity sensor or not when calling

Ported from CAF by @Sirhc95

Change-Id: I292e254ab8090d9fcb739109d29d96078779467e
CRs-Fixed: 914710
---
 java/com/android/incallui/ProximitySensor.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/java/com/android/incallui/ProximitySensor.java b/java/com/android/incallui/ProximitySensor.java
index cb87252ca..91291980a 100644
--- a/java/com/android/incallui/ProximitySensor.java
+++ b/java/com/android/incallui/ProximitySensor.java
@@ -49,6 +49,7 @@ public class ProximitySensor
     implements AccelerometerListener.OrientationListener, InCallStateListener, AudioModeListener, SensorEventListener {
 
   private static final String TAG = ProximitySensor.class.getSimpleName();
+  private static final String PROXIMITY_SENSOR = "proximity_sensor";
 
   private final PowerManager mPowerManager;
   private final PowerManager.WakeLock mProximityWakeLock;
@@ -278,6 +279,8 @@ public class ProximitySensor
             || CallAudioState.ROUTE_BLUETOOTH == audioRoute
             || mIsAttemptingVideoCall
             || mIsVideoCall);
+    screenOnImmediately |= Settings.System.getInt(mContext.getContentResolver(),
+                    PROXIMITY_SENSOR, 1) == 0;
 
     // We do not keep the screen off when the user is outside in-call screen and we are
     // horizontal, but we do not force it on when we become horizontal until the
